CMark.hsc

```diff
@@ -503,3 +591,24 @@ foreign import ccall "cmark.h cmark_node_set_on_exit"
 
 foreign import ccall "cmark.h &cmark_node_free"
     c_cmark_node_free :: FunPtr (NodePtr -> IO ())
+
+foreign import ccall "registry.h cmark_register_plugin"
+    c_cmark_register_plugin :: FunPtr (PluginPtr -> IO Int) -> IO ()
+
+foreign import ccall "core-extensions.h &core_extensions_registration"
+    c_core_extensions_registration :: FunPtr (PluginPtr -> IO Int)
+
+foreign import ccall "cmark_extension_api.h cmark_find_syntax_extension"
+    c_cmark_find_syntax_extension :: CString -> IO ExtensionPtr
+
+foreign import ccall "cmark.h cmark_llist_append"
+    c_cmark_llist_append :: MemPtr -> LlistPtr a -> Ptr () -> IO (LlistPtr a)
```

Comment:
I seriously didn't think putting in unbound type variables would Just Work™ here, but hey, it did! How neat!